src/gt4py/next/type_system/type_specifications.py

```diff
@@ -92,31 +79,37 @@ def __str__(self) -> str:
         return f"{kind_str}{self.shape}"
 
 
-@dataclass(frozen=True)
+class ListType(DataType):
+    element_type: DataType
+    # TODO(havogt): the `offset_type` is not yet used in type_inference,
+    # it is meant to describe the neighborhood (via the local dimension)
+    offset_type: Optional[common.Dimension] = None
```

Comment:
Why has this type been moved here / promoted to something shared between FOAST and ITIR type system?

Reply:
moved here because it's a possible dtype of FieldType, see your other comment...

Reply:
Makes sense, maybe a comment that this is not used in the frontend and only here because it can be in a field makes sense.